README.md

```diff
@@ -1,53 +1,60 @@
-       ------------------> |    DLX     |
-                           |            |         
-                            ------------   
-                                                         
-                                                          
-                                                         
-```
-
-### Autoscaler
-Based on custom metric name, will call `ResourceScaler` interface function `Scale` with a `Resource` from a list of known `Resource`'s (`GetResources` function)
-Config for service includes:
-```
-    Namespace     - kubernetes namespace of the resources
-    ScaleWindow   - an allowed period for the resource to be inactive before calling the `Scale` function
-    MetricName    - name of the metric to monitor for the resource
-    Threshold     - A threshold for a metric to concider the resource being inactive
-    ScaleInterval - an interval to call scale function which checks if a resource has passed the scale window, if not the scale window is being considered the longest duration of time where the metric for the resource is under the threshold
-```
-Scaler should change the resource service to point to DLX service and not to the scaled down pod (to be changed by `ResourceScaler` or some other external entity)
-
-### DLX
-A service that listens on http requests for a given namespace, when a request is received, it's being saved in-memory until `Scale` function returns, this is an indication that the resource is ready and any changes that were made by autoscaler were reverted (i.e. changing where service points to), that request is then being proxied to the new instance and the response from it returns to the user.
+# Scale to Zero
 
+Infrastructure to scale any resource to and from zero
+
+## Prerequisites
+
+**Custom metrics API implementation:**
+
+The Autoscaler takes decisions based on data queried from Kubernetes 
+[custom metrics API.](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/instrumentation/custom-metrics-api.md)
+There are several possible tools that implement it, we internally use
+[Prometheus](https://prometheus.io/) with the [Prometheus-Adapter](https://github.com/DirectXMan12/k8s-prometheus-adapter)
+but you can use which ever you want... you can find [here](https://github.com/kubernetes/metrics/blob/master/IMPLEMENTATIONS.md#custom-metrics-api) 
+recommended implementations
+
+## Design:
+![Design scheme](https://www.lucidchart.com/publicSegments/view/cc8927a6-537f-4fe6-95e1-731503bc7996/image.jpeg)
+
+**The autoscaler** - responsible to periodically check whether resources should be scale to zero, it does that by 
+querying the custom metrics API. When it decide a resource should be scaled to zero it tells the resource scaler to 
+scale the resource to zero.
+The resource scaler will route the traffic to the DLX, which in terms of K8s is done by changing the svc selector, 
+after that, it will scale the resource to zero.
```

Comment:
```suggestion
after doing so, it will perform the actual scaling to zero of the resource (by modifying a deployment / replicaset / daemon set / pods / etc).
```